### ISSUE TITLE
mixin-name-format should not warn on `@include`.

### ISSUE THREAD

**Issue by traviskaufman** [sasstools/sass-lint#739]
> What version of Sass Lint are you using?

1.7.0

> Please include any relevant parts of your configuration

```yaml
# .sass-lint.yml

options:
  formatter: stylish
files:
  include: _src/style/**/*.scss
rules:
  no-color-keywords: 0
  no-color-literals: 0
  no-extends: 1
  no-important: 0
  class-name-format:
    - 1
    -
      convention: hyphenatedbem
  border-zero:
    - 1
    -
      convention: none
  brace-style:
    - 1
    -
      allow-single-line: false
  property-sort-order:
    - 1
    -
      order: smacss
  variable-for-property: 0
```

> What did you do? Please include the actual source code causing the issue.

Using a mixin defined by a third-party sass library ([typesettings](https://github.com/ianrose/typesettings)) 
that defines mixins using camelCase, specifically `setType`.

> What did you expect to happen?

When I `@include setType()` I expected no sass-lint errors.

> What actually happened? Please include any error messages given to you by Sass Lint.

<img width="776" alt="screen shot 2016-06-12 at 12 44 57 pm" src="https://cloud.githubusercontent.com/assets/1185269/15992494/832659ee-309b-11e6-9dca-563d83f27b74.png">

It doesn't make a lot of sense to me to warn about mixin name formatting at a call site, mostly because if you're using a third-party library that doesn't adhere to your conventions, you're guaranteed to get false negatives. Instead, name formatting errors should only be surfaced at a definition site. 

> If you're using a IDE plugin have you tried the CLI too?

<img width="832" alt="screen shot 2016-06-12 at 12 48 03 pm" src="https://cloud.githubusercontent.com/assets/1185269/15992521/f82ac1a8-309b-11e6-95dd-0dedd17bd931.png">

**Comment by wawjr3d** [comment]
@traviskaufman what was your workaround for this problem? i have the same issue
